### PR TITLE
Namespace mongoengine-specific stuff in flask_common.mongo

### PR DESCRIPTION
**Pull request by wojcikstefan**
This PR puts (almost) everything that's specific to MongoDB/MongoEngine in `flask_common.mongo.*` submodules.

This is first and foremost a better structure, but it'll also make it easier to understand a) what needs to change when we move to upstream MongoEngine, b) where to put anything else that's mongo-specific (I may soon add some code in here if it won't make it to upstream MongoEngine).

Probably best to review the commits separately. You may also want to check out this branch and check the code structure in your editor.

### CODE REVIEW

File: tests/test_documents.py [hunk 1]
```diff
@@ -1,4 +0,0 @@
-def test_documents():
-    from flask_common import documents
-    # Just making sure import works
```
Comment (tsx) [inline tests/test_documents.py]: I guess my fault for under-documeting this. The intention is to syntax-check it against python3. Even if the code isn't fully compatible yet, it makes sure that py3 can at least parse it without failing with SyntaxError (old `raise a,b,c` syntax for example)

File: flask_common/mongo/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from .documents import DocumentBase, RandomPKDocument, SoftDeleteDocument
```
Comment (tsx) [inline flask_common/mongo/__init__.py]: I'd prefer not to re-export it. This package doesn't result in a better or higher-level API then its "internals", so `.mongo` becomes just a pile of unrelated stuff.

File: flask_common/mongo/__init__.py [hunk 3]
```diff
@@ -1,4 +1,27 @@
 from .documents import DocumentBase, RandomPKDocument, SoftDeleteDocument
+from .fields import LowerEmailField, LowerStringField, TrimmedStringField
 from .querysets import ForbiddenQueryException, ForbiddenQueriesQuerySet
 from .query_counters import custom_query_counter
 from .utils import fetch_related, iter_no_cache
+
+# Import fields that exist only if some dependencies are installed.
+
+try:
+    from .fields import EncryptedStringField
```
Comment (tsx) [inline flask_common/mongo/__init__.py]: Since the idea of this PR is to isolate all mongo-specific stuff into a separate package, I don't see value of skipping those anymore.

If client code don't use mongo, they have no business even trying to import _anything_ from `flask_common.mongo`. And if they do use mongo and have issues with it (for example wrong version), the try-except would swallow the root cause of any actual issues they might have - but would fail _anyway_. All the client would see is `cannot import name EncryptedStringField` instead what's really wrong.

File: flask_common/mongo/__init__.py (hunk 2 above)
Comment (wojcikstefan) [reply flask_common/mongo/__init__.py]: Good call. Done in ca3abc5.

File: flask_common/mongo/__init__.py (hunk 3 above)
Comment (wojcikstefan) [reply flask_common/mongo/__init__.py]: I don't think these specific fields are skipped because of MongoEngine. In this particular case, I believe the field is skipped so that we can use `flask_common` w/o installing a heavy `pycrypto` package.

File: tests/test_documents.py (hunk 1 above)
Comment (wojcikstefan) [reply tests/test_documents.py]: Aha, gotcha. Brought the test back in 52ad89d.
